cupyx/scipy/sparse/sputils.py

```diff
@@ -115,3 +116,54 @@ def upcast(*args):
             return t
 
     raise TypeError('no supported conversion for types: %r' % (args,))
+
+
+def check_shape(args, current_shape=None):
+    """Check validity of the shape"""
+
+    if len(args) == 0:
+        raise TypeError("function missing ! required positional argument: "
+                        "'shape'")
+
+    elif len(args) == 1:
+        try:
+            shape_iter = iter(args[0])
+        except TypeError:
+            new_shape = (operator.index(args[0]), )
+        else:
+            new_shape = tuple(operator.index(arg) for arg in shape_iter)
+    else:
+        new_shape = tuple(operator.index(arg) for arg in args)
+
+    if current_shape is None:
+        if len(new_shape) != 2:
+            raise ValueError('shape must be a 2-tuple of positive integers')
+        elif new_shape[0] < 0 or new_shape[1] < 0:
+            raise ValueError("'shape' elements cannot be negative")
+
+    else:
+        current_size = cupy.prod(cupy.asarray(current_shape))
```

Comment:
`current_shape` is a small length-2 tuple. Computing it on host with numpy is far more efficient than doing the same on GPU.